### PR TITLE
AWS sagemaker: fixed a bug in ground_truth and updated all components to use images from new docker hub repo

### PR DESCRIPTION
**Pull request by akartsky**
AWS ground_truth component expects active_learning_model_arn.txt to be always non-empty 
[component.yaml #L129](https://github.com/kubeflow/pipelines/blob/cfcd824ed1c7bb6924cee30b14e49d281f1d4a73/components/aws/sagemaker/ground_truth/component.yaml#L129)

But when we don't pass the optional argument "enable_auto_labeling", empty string is stored in [active_learning_model_arn.txt](https://github.com/kubeflow/pipelines/blob/cfcd824ed1c7bb6924cee30b14e49d281f1d4a73/components/aws/sagemaker/ground_truth/src/ground_truth.py#L71)

- [string variable which gets stored in active_learning_model_arn.txt](https://github.com/kubeflow/pipelines/blob/cfcd824ed1c7bb6924cee30b14e49d281f1d4a73/components/aws/sagemaker/ground_truth/src/ground_truth.py#L64) 
- [util function which returns that value](https://github.com/kubeflow/pipelines/blob/cfcd824ed1c7bb6924cee30b14e49d281f1d4a73/components/aws/sagemaker/common/_utils.py#L804)

Solved the issue by storing space instead of empty string 

Error message: 
```
This step is in Error state with this message: failed to save outputs: 
path /tmp/active_learning_model_arn.txt does not exist (or /tmp/active_learning_model_arn.txt is empty)
```
----------

uploaded this change to new docker hub repo 
and
updated readme for ground_truth_pipeline_demo pipeline

### CODE REVIEW

**Comment by Jeffwan** [comment]
@akartsky Is this a stable version? Do you want to correct tag in this PR?

File: samples/contrib/aws-samples/ground_truth_pipeline_demo/README.md [hunk 1]
```diff
@@ -66,11 +70,14 @@ Open the Kubeflow pipelines UI. Create a new pipeline, and then upload the compi
 
 The pipeline requires several arguments - replace `role_arn`, Amazon Cognito information, and the S3 input paths with your settings, and run the pipeline.
 
+> Note : team_name, ground_truth_train_job_name and ground_truth_validation_job_name need to be unique or else pipeline will error out if the names already exist
```
Comment (surajkota) [inline samples/contrib/aws-samples/ground_truth_pipeline_demo/README.md:73]: unique for a region? or account ? or globally like S3 ?

File: samples/contrib/aws-samples/ground_truth_pipeline_demo/README.md [hunk 2]
```diff
@@ -18,18 +18,21 @@ Create a s3 bucket and run [this python script](https://github.com/kubeflow/pipe
...
-If you already have private workteams in the region you are using, skip step 1 and do steps 2 and 3 in the user pool that is already being used.
+For this demo you can create a new user pool (if you don't have one already).   
 
-Steps:
-1. Create a user pool in Amazon Cognito. Configure the user pool as needed, and make sure to create an app client. The Pool ID will be found under General settings.
-2. After creating the user pool, go to the Users and Groups section and create a group. Create users for the team, and add those users to the group.
-3. Under App integration > Domain name, create an Amazon Cognito domain for the user pool.
+AWS console -> Amazon SageMaker -> Ground Truth, Labeling workforces -> Private -> Create Private Team -> Give it "KFP-ground-truth-demo-pool" name and use your email address -> Create Private team -> Click on the radio button and from summary note down the "Amazon Cognito user pool", "App client" and "Labeling portal sign-in URL" -> click on the team name that you created and note down "Amazon Cognito user group"
```
Comment (surajkota) [inline samples/contrib/aws-samples/ground_truth_pipeline_demo/README.md:28]: Can we make this better by using a blog or aws docs as reference for this process? 

If we don't have an existing reference, can you do something like [this link](https://docs.aws.amazon.com/elastic-inference/latest/developerguide/setting-up-ei.html#eia-launch) which guides user on console process to launch an instance with EI

File: samples/contrib/aws-samples/ground_truth_pipeline_demo/README.md [hunk 3]
```diff
@@ -56,6 +59,7 @@ data:
 
 Follow the guide to [building a pipeline](https://www.kubeflow.org/docs/guides/pipelines/build-pipeline/) to install the Kubeflow Pipelines SDK, then run the following command to compile the sample Python into a workflow specification. The specification takes the form of a YAML file compressed into a `.tar.gz` file.
 
+
```
Comment (surajkota) [inline samples/contrib/aws-samples/ground_truth_pipeline_demo/README.md:62]: nit: double new line ?

**Comment by akartsky** [comment]
> @akartsky Is this a stable version? Do you want to correct tag in this PR?

done

**Pull request merged**